### PR TITLE
add ipumspy package

### PR DESCRIPTION
**Pull request by renae-r**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by renae-r** [comment]
The github user named in the maintainers is actually an organization. I am a member of that organization and will be the primary maintainer for now.

**Comment by renae-r** [comment]
@conda-forge-admin, please ping team

I know the conda-forge team is very busy - just wanted to check in on this and make sure there wasn't a problem with my PR! Thanks

File: recipes/ipumspy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: e64249ba3e22a84a99ded0d630dc5977c7b5d464a4074a8407c39c3f2ccc4011
+
+build:
+  number: 0
+  entry_points:
+    - ipums = ipumspy.cli:cli
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py2k]
```
Comment (ocefpaf) [inline recipes/ipumspy/meta.yaml:17]: You don't need this skip b/c we don't build py2k. Also, you probably can make this one noarch. You'll have to drop the py38 selectors below.

Comment (renae-r) [reply recipes/ipumspy/meta.yaml]: Thanks @ocefpaf , I have addressed these.

File: recipes/ipumspy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+requirements:
+  host:
+    - click >=7.0.0,<8.0.0
+    - pandas >=1.0.4,<2.0.0
+    - pip
+    - pyarrow >=3.0.0,<4.0.0
+    - python >=3.6.2
+    - pyyaml >=5.4.1,<6.0.0
+    - requests >=2.25.1,<3.0.0
+    - importlib-metadata >=4.0.1
```
Comment (ocefpaf) [inline recipes/ipumspy/meta.yaml:28]: You only need all the run dependencies at build time if the package imports itself when building. Is that the case here? If not you can remove all the run dependencies from `host`.

Comment (renae-r) [reply recipes/ipumspy/meta.yaml:28]: @ocefpaf my apologies for the very delayed response. I am not familiar enough with the build process to know whether the package imports itself when building. Certain modules import from other modules within the library at runtime - does this mean the package imports itself when building?

Comment (ocefpaf) [reply recipes/ipumspy/meta.yaml:28]: Only if it import itself in setup.py. The best way to figure it out is to remove those from here. Let's give it a try.

Comment (renae-r) [reply recipes/ipumspy/meta.yaml]: @ocefpaf I removed the run dependencies from the `host` element and was able to build locally and install. Latest commit has that change. Thanks!

**Pull request merged**